
<launch>

  <arg name="camera_info_url" default="cam_c270"/>
  <arg name="rgbcameraframe" default="rgb_camera_frame" />
  <arg name="viewimage" default="false" />
  <arg name="device" default="/dev/video0" />
  <arg name="format" default="yuyv" />  <!-- yuyv, mjpeg -->
  <arg name="framerate" default="15"/>
  <arg name="io_method" default="mmap"/>
  <arg name="contrast" default="255"/>
  <arg name="camera_frame_id" default="camera_link"/>

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="log" >
    <param name="video_device" value="$(arg device)" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="$(arg format)" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="$(arg io_method)"/>
    <param name="framerate" value="$(arg framerate)"/>
    <param name="contrast" value="$(arg contrast)"/>
    
    <param name="camera_info_url" value=" file://$(find marrtino_r3d)/config/$(arg camera_info_url).yaml"/>
   <!-- <param name="camera_info_url" value=" file://$(find usb_cam)/cam_paolo.yaml"/> -->
     
  </node>
  <group if="$(arg viewimage)">
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>
 </group>

</launch>
